extensions/azdata/src/azdata.ts

```diff
@@ -145,21 +146,111 @@ export async function upgradeAzdata(outputChannel: vscode.OutputChannel): Promis
 }
 
 /**
+ * Checks whether azdata is installed - and if it is not then invokes the process of azdata installation.
+ * @param outputChannel Channel used to display diagnostic information
+ * @param userRequested true means that this operation by was requested by a user by executing an ads command.
+ */
+export async function checkAndInstallAzdata(outputChannel: vscode.OutputChannel, userRequested: boolean = false): Promise<IAzdataTool | undefined> {
+	try {
+		const azdata = await findAzdata(outputChannel); // find currently installed Azdata
+		// Don't block on this since we want the extension to finish activating without needing user input
+		checkAndUpgradeAzdata(azdata, outputChannel, userRequested).catch(err => {
+			//update if one is available and user wants it.
+			vscode.window.showWarningMessage(loc.updateError(err));
+			outputChannel.appendLine(loc.updateError(err));
+		}).then(() => {
+			return findAzdata(outputChannel); // now again find and return the currently installed azdata
```

Comment:
This is returning from the lambda function so won't do anything. You should be awaiting the call to checkAndUpgradeAzdata and then returning findAzdata. 

The general idea I think we should be going for is that all these helper methods should be fully blocking async functions which will only return once azdata is installed/upgraded. And then in the extension activate we'll just use .then instead of blocking on the calls to these to ensure that extension activation isn't impacted by user input.

Reply:
ok